restock.lic

```diff
@@ -0,0 +1,107 @@
+#   Documentation: https://elanthipedia.play.net/Lich_script_repository#restock
+
+custom_require.call(%w[common common-items common-money])
+
+class Restock
+  include DRC
+  include DRCM
+  include DRCI
+
+  def initialize
+    setup
+
+    items = parse_restockable_items
+    items_to_restock = []
+    coin_needed = 0
+
+    items.each do |item|
+      remaining = if item['stackable']
+                    count_stackable_item(item['name'])
+                  else
+                    count_nonstackable_item(item['name'])
+                  end
+
+      next unless remaining < item['quantity']
+      num_needed = item['quantity'] - remaining
+      buy_num = (num_needed / item['size'].to_f).ceil
+      coin_needed += buy_num * item['price']
+      item['buy_num'] = buy_num
+      items_to_restock.push(item)
+    end
+
+    ensure_copper_on_hand(coin_needed, @hometown) if coin_needed > 0
```

Comment:
Do you need this to be conditional? I thought `ensure_copper_on_hand` would take care of that

Reply:
I just wanted to avoid the 'wealth' command in the event that there was nothing needing to be restocked.